### PR TITLE
ByPass the FileDetector for things that are not files

### PR DESCRIPTION
**Pull request by jtnord**
The FileDetector replaces any string that is also a local file by a
temp string and uploads that file to the remote system (at the location specified in the temp string).

This is generally a good thing, but sometimes the string that is also a file path is really a string and not something local or that needs replaces.

This is always the case for the JVM path when used in an external system.

Amends #780   detected when trying to bump this as a dependency in a closed source test suite.

```
[08/09/22 12:54:32] [SSH] Starting agent process: cd "/home/test" && /tmp/7fd87797-fc61-4f70-9b2d-079c37ac4a49/upload1745090074472040540file/java  -jar remoting.jar -workDir /home/test -jar-cache /home/test/remoting/jarCache
sh: 1: /tmp/7fd87797-fc61-4f70-9b2d-079c37ac4a49/upload1745090074472040540file/java: not found
Agent JVM has terminated. Exit code=127
[08/09/22 12:54:32] Launch failed - cleaning up connection
[08/09/22 12:54:32] [SSH] Connection closed.
```

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java [hunk 1]
```diff
@@ -24,7 +24,7 @@ public class SshSlaveLauncher extends ComputerLauncher {
     public final Control suffixCmd = control("suffixStartSlaveCmd");
     public final Control timeout = control("launchTimeoutSeconds");
     public final Control retries = control("maxNumRetries");
-    public final Control javaPath = control("javaPath");
+    private final Control javaPath = control("javaPath");
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java:27]: all the controlls should really be provate and hidden behind get/set but that is out of scope in this change

File: src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java [hunk 2]
```diff
@@ -44,6 +44,11 @@ public SshCredentialDialog addCredential() {
         return new SshCredentialDialog(getPage(), "/credentials");
     }
 
+    public void setJavaPath(String jvmPath) {
+        try (UselessFileDetectorReplacement ufd = new UselessFileDetectorReplacement(driver)) {
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java:48]: if anything else uses `path`ish like Strings they would need to be similarly wrapped, however no other code has been observed to fail yet.

File: src/main/java/org/jenkinsci/test/acceptance/selenium/UselessFileDetectorReplacement.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    private final RemoteWebDriver remoteDriver;
+
+    /** 
+     * Create a new instance suitable for use in a try-with-resources block.
+     * Example usage is:<code>
+     * try (UselessFileDetectorReplacement ufd = new UselessFileDetectorReplacement(driver)) {
+     *   control.setText(someTextThatIsAlsoALocalFileName);
+     * }
+     * </code>
+     * @param driver
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/test/acceptance/selenium/UselessFileDetectorReplacement.java:25]:
```suggestion
```

**Review by timja — APPROVED** [review]

**Comment by jtnord** [comment]
Does anyone know if all these failures are due to #877 (given there is not a run on `master` to compare against I am trying to look at a few PRs...  they all seem relevant.

If so I would like to merge and release this to unblock some other work - given this is not causing any new regressions.

**Comment by NotMyFault** [comment]
> given there is not a run on master

I was going ahead and give it a try a couple of days ago in https://github.com/jenkinsci/acceptance-test-harness/pull/882, but the outcome would be the same one tracked in https://github.com/jenkinsci/acceptance-test-harness/issues/877.

https://github.com/jenkinsci/acceptance-test-harness/pull/869 shows the same results 😢

**Review by NotMyFault — APPROVED** [review]
